specs/language/declarations.tex

```diff
@@ -56,6 +56,74 @@
 \p If a function is declared with an \texttt{export} specifier then all redeclarations of the same function must also use the \texttt{export} specifier or be part of \textit{export-declaration-group} (\ref{Decl.Export}).
 
 \Sec{Declarators}{Decl.Decl}
+
+\begin{grammar}
+  \define{init-declarator-list}\br
+  \textit{init-declarator}\br
+  \textit{init-declarator-list} \terminal{,} \textit{init-declarator}\br
+
+  \define{init-declarator}\br
+  \textit{declarator} \opt{initializer}\br
+
+  \define{declarator}\br
+  \textit{declarator-id} \opt{attribute-specifier-seq}\br
+  \textit{declarator} \textit{parameters-and-qualifiers}\br
+  \textit{declarator} \terminal{[} \opt{constant-expression} \terminal{]} \opt{attribute-specifier-seq}\br
+
+  \define{parameters-and-qualifiers}\br
+  \terminal{(} \textit{parameter-declaration-clause} \terminal{)} \opt{cv-qualifier-seq} \opt{attribute-specifier-seq}\br
+
+  \define{cv-qualifier-seq}\br
+  \textit{cv-qualifier} \opt{cv-qualifier-seq}\br
+
+  \define{cv-qualifier}\br
+  \terminal{const}\br
+
+  \define{declarator-id}\br
+  \textit{id-expression}\br
+
+\end{grammar}
+
+\Sub{Arrays}{Decl.arrays}
+% This might need to nest deeper as we flesh out the sections.
+
+\p A declaration \textit{T D} where \textit{D} is of the form
+
+\begin{grammar}
+\textit{D`} \terminal{[} \textit{N} \terminal{]} \opt{attribute-specifier-seq}\br
+\end{grammar}
+
+declares an array of \textit{N} \textit{T`}, where \textit{N} is the
+\textit{array bound}, and \textit{T`} is the derived type of the declarator
+\textit{D`} and is called the \textit{contained type} of the array. \textit{N}
+shall be a constant expression converted to unsigned integer type and shall have
+a value greater than zero.
+
+\p A declaration \textit{T D} where \textit{D} is of the form
+
+\begin{grammar}
+\textit{D`} \terminal{[} \terminal{]} \opt{attribute-specifier-seq}\br
+\end{grammar}
+
+declares an array of unknown bound, except when the bound is implicitly derived
+as described below. A declaration may only declare an array of unknown bound for
+global declarations of resource objects (\ref{Resources}).
+
+\p The contained type of an array of either form shall be a complete type.
+
+\begin{note}
+  An array of arrays may be declared with infinite nesting of array declarators
+  as long as either all array bounds are known, or only the outermost array
+  bound is unknown. This is codified in the spec above with the requirement that
+  the contained type of an array be a complete type.
+\end{note}
+
+\p A declarator with an omitted array bound is valid if the when declaring an
+object of array type which is not a non-static data member and is initialized
+with an initializer that follows the declarator. In this case the array bound is
```

Comment:
this is only allowed for global declarations of resource objects? or for all array declarations?